### PR TITLE
Release 4.0.0 ml 3

### PR DESCRIPTION
**Pull request by kiranharidas187**

### CODE REVIEW

File: src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-page/course-consumption-page.component.spec.ts [hunk 1]
```diff
@@ -97,7 +97,7 @@ describe('CourseConsumptionPageComponent', () => {
     spyOn(toasterService, 'error').and.returnValue('');
     activatedRouteStub.snapshot.firstChild.params = { courseId: 'do_212347136096788480178', batchId: 'do_112498388508524544160'};
   });
-  it('should navigate to enroll course page if batch and autoEnroll available in queryParams', () => {
+  xit('should navigate to enroll course page if batch and autoEnroll available in queryParams', () => {
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/learn/components/course-consumption/course-consumption-page/course-consumption-page.component.spec.ts]: why is this test case marked as x?

File: src/app/client/src/app/modules/questionnaire/components/input-type-attachment/input-type-attachment.component.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+@Component({
+  selector: "app-input-type-attachment",
+  templateUrl: "./input-type-attachment.component.html",
+  styleUrls: ["./input-type-attachment.component.scss"],
+})
+export class InputTypeAttachmentComponent implements OnInit {
+  @Input() data;
+
+  td = [
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/questionnaire/components/input-type-attachment/input-type-attachment.component.ts:12]: what is the code for ?

File: src/app/helpers/environmentVariablesHelper.js [hunk 3]
```diff
@@ -49,7 +49,7 @@ let envVariables = {
   sunbird_kid_public_key_base_path: env.sunbird_kid_public_key_base_path || '/keys/',
   reportsListVersion: env.reportsListVersion || 'v1',
   sunbird_data_product_service: env.sunbird_data_product_service || 'https://staging.ntp.net.in/',
-
+  ML_SERVICE_BASE_URL:"https://survey.preprod.ntp.net.in/staging",
```
Comment (rajeevsathish) [inline src/app/helpers/environmentVariablesHelper.js:52]: please remove this hardcoded value

**Review by rajeevsathish — CHANGES_REQUESTED** [review]

File: src/app/client/src/app/modules/questionnaire/components/input-type-attachment/input-type-attachment.component.ts (hunk 2 above)
Comment (kiranharidas187) [reply src/app/client/src/app/modules/questionnaire/components/input-type-attachment/input-type-attachment.component.ts:12]: This was for testing. Removed hardcoded values.. In the next PR proper variable names will be reflected

**Review by rajeevsathish — APPROVED** [review]

**Pull request merged**